src/liquid/components/ld-sidenav/ld-sidenav.tsx

```diff
@@ -448,7 +449,20 @@ export class LdSidenav {
     this.collapsed = !this.collapsed
   }
 
-  private activeSubnavContainsIcons = () => {
+  private activeSubnavContainsIconsOrHasBack = () => {
+    // Return true if we have a back button or a nav items as a direct child.
+    if (
+      Array.from(this.el.children).filter(
+        (child) =>
+          child.tagName === 'LD-SIDENAV-BACK' ||
+          (child.tagName === 'LD-SIDENAV-NAVITEM' &&
+            !(child as HTMLLdSidenavNavitemElement).mode)
+      ).length
```

Comment:
I kind of prefer to have an explicit boolean expression e.g.
```suggestion
      ).length > 0
```